crates/red_knot/src/semantic/types/infer.rs

```diff
@@ -608,4 +608,22 @@ mod tests {
 
         assert_public_type(&case, "a", "a", "(Literal[1] | Literal[2])")
     }
+
+    #[test]
+    fn ifexpr_nested() -> anyhow::Result<()> {
+        let case = create_test()?;
+
+        write_to_path(
+            &case,
+            "a.py",
+            "
+                class C1: pass
+                class C2: pass
+                class C3: pass
+                x = C1 if flag else C2 if flag2 else C3
+            ",
+        )?;
+
+        assert_public_type(&case, "a", "x", "(Literal[C1] | Literal[C2] | Literal[C3])")
```

Comment:
When printing the revealed type to the user, it would be nice if we could prettify this to `Literal[C1, C2, C3]` (pyright does this: https://pyright-play.net/?pyrightVersion=1.1.361&strict=true&code=MYRgBAvGIFDATJM84GYmpgMwDYEMBzAGjF0MSgCMB7anACgEoSa6mYAPJUMASy1L4CYAKY4AziLAI%2BAsgURjJ0zACcRANxF4cAfQAuATwAOI%2Bh0YwgA). But doesn't need to be done now

Reply:
Oh, I see you already created #11782 :)

Reply:
Feel free to open an issue for this!

One thing that's not clear to me is whether we want to mix different kinds of literals or not. E.g. do we prefer `Literal[1, C, "foo", "bar"]` or `Literal[1] | Literal[C] | Literal["foo", "bar"]`?

Reply:
In #11784 I did this condensing, but only for int literals so far.

Reply:
> One thing that's not clear to me is whether we want to mix different kinds of literals or not. E.g. do we prefer `Literal[1, C, "foo", "bar"]` or `Literal[1] | Literal[C] | Literal["foo", "bar"]`?

yeah, not sure... especially with objects (such as classes) that users might not be expecting to see in a `Literal` slice... I'll possibly open an issue tomorrow, gonna call it for the night pretty soon!

Reply:
I went ahead and created #11785